File: reacciones/solutions/correct/ralonso.py
from collections import defaultdict

counts = defaultdict(lambda: 0, {})

n, k = (int(i) for i in input().split())

message = input()

results = []

for i in message:
    counts[i] += 1
    if counts[i] > k:
        print("imposible")
        exit(0)
    results.append(counts[i])

print(" ".join(str(result) for result in results))

File: reacciones/solutions/correct/ralonso.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(int argc, char *argv[]) {
  vector<int> counts(26);

  int n, k;
  string message;

  cin >> n >> k;
  cin >> message;

  vector<int> output(n);

  for (int i = 0; i < n; ++i) {
    char c = message[i];

    if (++counts[c - 'a'] > k) {
      cout << "imposible" << endl;
      return 0;
    }

    output[i] = counts[c - 'a'];
  }

  for (int i = 0; i < n; ++i) {
    cout << output[i];
    if (i == n - 1) {
      cout << endl;
    } else {
      cout << ' ';
    }
  }
  return 0;
}
